### PR TITLE
Python: Fixed type error in plan.run_next_step_async method

### LINKED ISSUE #3219: Python: TypeError: Kernel.create_new_context() takes 1 positional argument but 2 were given
**Describe the bug**
In the python sdk (0.3.12.dev0), when running `Plan.run_next_step_async` the method throws a TypeError. Inside the method, there's a call to `Kernel.create_new_context`. The `create_new_context` method does not take any arguments, but ContextVariables are passed in.

Here's the implementation of Plan.run_next_step_async:

```python
async def run_next_step_async(
        self,
        kernel: Kernel,
        variables: ContextVariables,
    ) -> "Plan":
        context = kernel.create_new_context(variables)
        return await self.invoke_next_step(context)
```

and here's the implementation of `Kernel.create_new_context`:

```python
def create_new_context(self) -> SKContext:
        return SKContext(
            ContextVariables(),
            self._memory,
            self.skills,
            self._log,
        )
```

**To Reproduce**
1. instantiate a kernel
2. create a plan
3. run `plan.run_next_async(kernel, ContextVariables())`

```python
import semantic_kernel as sk
from semantic_kernel.planning.plan import Plan
from semantic_kernel.orchestration.context_variables import ContextVariables

kernel = sk.Kernel()
plan = Plan()
plan = await plan.run_next_step_async(kernel, ContextVariables())
```

**Expected behavior**
The method should not raise a `TypeError`

**Screenshots**

**Platform**
 - OS: Mac
 - IDE: Intellij
 - Language: Python
 - Source: 0.3.12.dev0

**Additional context**
The fix seems to be one of the two options:

1. instantiate `SKContext` directly instead of using the kernel
2. create another method on `Kernel` to create the `SKContext` while passing in `ContextVariables`

I'm happy to submit a PR to resolve this, but need some help in the overall direction of the fix.

### PR DESCRIPTION
**Pull request by AdarshAcharya5**
### Description

This PR fixes #3219 

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/planning/plan.py [hunk 1]
```diff
@@ -290,7 +290,8 @@ async def run_next_step_async(
         kernel: Kernel,
         variables: ContextVariables,
     ) -> "Plan":
-        context = kernel.create_new_context(variables)
+        context = kernel.create_new_context()
+        context.variables = variables
```
[the remark this answers is not in the document]
Comment (AdarshAcharya5) [reply python/semantic_kernel/planning/plan.py:294]: @dmytrostruk Absolutely. I had done this initially!, but in many other files, create_new_context() method had been used without context variables, so I assumed that the method was only being used to create a fresh instance of context object.

Screenshot (490)~2

I will make the change you mentioned right now!.

[the remark this answers is not in the document]
Comment (AdarshAcharya5) [reply python/semantic_kernel/planning/plan.py]: @dmytrostruk I have made the changes. Please let me know if there needs to be any other change!

[the remark this answers is not in the document]
Comment (dmytrostruk) [reply python/semantic_kernel/planning/plan.py]: @AdarshAcharya5 thanks a lot, looks good to me! 

It seems like there is formatting issue:
image

You can configure Black formatter in your IDE to format code on file save or format/check formatting with following command:
`poetry run black <file_path> --check`

[the remark this answers is not in the document]
Comment (AdarshAcharya5) [reply python/semantic_kernel/planning/plan.py]: Apologies!. Should work now

[the remark this answers is not in the document]
Comment (dmytrostruk) [reply python/semantic_kernel/planning/plan.py]: LGTM, thanks a lot!

**Review by dmytrostruk — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**